src/Paket.Core/Versioning/SemVer.fs

```diff
@@ -96,19 +132,18 @@ type SemVerInfo =
       /// The optional PreRelease version
       PreRelease : PreRelease option
       /// The optional build no.
-      Build : string
+      Build : bigint
```

Comment:
there was no case in unit or integration tests actually using string builds; though they could have been created by accident (we actually did)

Reply:
I think the compelling case for `string` here instead of `bigint` is the case of embedding commit hashes as the build metadata. The spec (section 10) says that build metadata can have the following form:
> ...Identifiers MUST comprise only ASCII alphanumerics and hyphen [0-9A-Za-z-]...

and so I think we must allow string here.

Reply:
As far as I understand it, build metadata is separate field, a semver compliant one, placed after a `+` sign, while the `Build` field is the 4th optional, non-semver, but NuGet supported, numeric segment, separated by dot, and having no meaningful comparison semantics for strings.

Reply:
ah, of course you are correct